spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -911,6 +935,14 @@ public void testPartitionedImportFromEmptyPartitionDoesNotThrow() {
     new StructField("ts", DataTypes.DateType, true, Metadata.empty())
   };
 
+  private static final StructField[] dateHourStruct = {
+    new StructField("id", DataTypes.IntegerType, true, Metadata.empty()),
+    new StructField("name", DataTypes.StringType, true, Metadata.empty()),
+    new StructField("dept", DataTypes.StringType, true, Metadata.empty()),
+    new StructField("ts", DataTypes.DateType, true, Metadata.empty()),
+    new StructField("hour", DataTypes.StringType, true, Metadata.empty())
```

Comment:
Question, (maybe I am not understanding original problem entirely), can we re-use existing test struct, ie, using "dept=01" as partition field?

Reply:
I wanted to reproduce the failure with date/hour, even if I use dept I will still have to keep some of the code for creating  table with different partition.

will keep it as is for readability and to have the reproducible test which is relatable with date/hour

Reply:
OK in that case, I'd prefer not extra structs that's not strictly necessary, to keep the changes smaller.  I dont see string hour being of a value like 01 being much more readable than a dept that has a name like 01 to justify a new struct.

I think we can still make a separate DF if we need to.

Reply:
Done